MusicTaste/MusicTaste/APIManagers/SpotifyAPIManager.m

```diff
@@ -156,35 +158,51 @@ -(void) saveSpotifyData:(NSDictionary *) spotifyData{
     PFUser *current = [PFUser currentUser];
     PFQuery *query = [PFQuery queryWithClassName:@"Music"];
     [query findObjectsInBackgroundWithBlock:^(NSArray *results, NSError *error) {
-        if (![[results valueForKey:@"userId"]containsObject:current.objectId]){
-            //Checks if User has already been added to database before adding
-            music[@"userId"] = current.objectId;
+        [User user].artists = [NSArray arrayWithArray:spotifyData[@"images"]];
+        if (![[[results valueForKey:@"author"] valueForKey:@"objectId"] containsObject: current.objectId]){
+            music[@"author"] = current;
             music[@"genres"] = [NSArray arrayWithArray:spotifyData[@"genres"]];
             music[@"tracks"] = [NSArray arrayWithArray:spotifyData[@"tracks"]];
             music[@"artists"] = [NSArray arrayWithArray:spotifyData[@"artists"]];
             music[@"albums"] = [NSArray arrayWithArray:spotifyData[@"albums"]];
             music[@"images"] = [NSArray arrayWithArray:spotifyData[@"images"]];
-            music[@"userimage"] = self.profilePicture;
-            music[@"username"] = self.name;
+            [self getSpotifyData:@"https://api.spotify.com/v1/me" completion:^(NSDictionary * userDict, NSError * error) {
+                if (!error){
+                    PFUser *current = [PFUser currentUser];
+                    NSString *name = userDict[@"display_name"];
+                    NSArray *images = userDict[@"images"];
+                    NSDictionary *imageDict = [images objectAtIndex:0];
+                    if (imageDict != nil){
+                        NSString * url =[NSString stringWithFormat:imageDict[@"url"], nil];
+                        music[@"username"] = [NSString stringWithFormat: name, nil];
+                        music[@"userimage"] = url;
+                        [music saveEventually];
+                    }
+                }
+            }];
             [music saveEventually];
         }
+        for (PFUser *user in results){
+            if ([[[user valueForKey:@"author"] valueForKey:@"objectId"] isEqual:current.objectId] )
+                self.author = user;
+        }
+        [[NSNotificationCenter defaultCenter]
```

Comment:
Send notification to MatchmakingViewController once the callback is completed

Reply:
This feels like new functionality that should be in the pull request description.